src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationManager.cs

```diff
@@ -63,35 +59,24 @@ public ConfigurationManager()
         /// <inheritdoc/>
         public IEnumerable<IConfigurationSection> GetChildren()
         {
-            lock (_providerLock)
-            {
-                // ToList() to eagerly evaluate inside lock.
-                return this.GetChildrenImplementation(null).ToList();
-            }
+            using var reference = _providerManager.GetReference();
+            return this.GetChildrenImplementation(reference.Providers, path: null);
         }
 
         IDictionary<string, object> IConfigurationBuilder.Properties => _properties;
 
         IList<IConfigurationSource> IConfigurationBuilder.Sources => _sources;
 
-        IEnumerable<IConfigurationProvider> IConfigurationRoot.Providers
-        {
-            get
-            {
-                lock (_providerLock)
-                {
-                    return new List<IConfigurationProvider>(_providers);
-                }
-            }
-        }
+        // We cannot track the duration of the reference to the providers if this property is used.
+        // If a configuration source is removed after this is accessed but before it's completely enumerated,
+        // this may allow access to a disposed provider.
```

Comment:
How concerned should we be able this scenario?

Reply:
I definitely don't like it, but I'm not very concerned because this comment is describing the existing behavior and I haven't heard complaints yet. Even when we had the `_providerLock`, we released it before returning the copied list.

I can think of two main reasons this hasn't been a bigger issue:
1. It's not super common to access `IConfigurationRoot.Providers` vs other `IConfiguration` methods and accessors.
2. Disposing IConfigurationProviders often doesn't render them completely unusable but instead just unregisters from file change notifications or similar.